burn-core/src/optim/sgd.rs

```diff
@@ -57,6 +58,7 @@ impl<B: Backend> SimpleOptimizer<B> for Sgd<B> {
         tensor: Tensor<B, D>,
         mut grad: Tensor<B, D>,
         state: Option<Self::State<D>>,
+        gradient_clip: Option<GradientClipper>,
```

Comment:
I'm not sure if you should put gradient clipper in the optimizer or in the training loop. What is your opinion on this?

Reply:
That's a good point. With TensorFlow I know that you generally pass in your gradient clipping values to your optimizer with something like `optimizer = SGD(lr=0.01, momentum=0.9, clipvalue=1.0)`, at least if you're using Keras. With PyTorch though I think it's in the training loop. 

Attaching it the optimizer seems to be more simple and allows users to do a little less setup in the training loop. Also making it part of the optimizer directly makes things a bit more consistent during optimization. 

On the other hand, making it a separate function called during the training loop makes it more flexible and allows you to control when it's applied. So that could be more useful when working with highly customized or complicated training loops that need very specific tuning. 

Not sure what you're preference is. We can also just do both lol

Reply:
If we do it in the optimizer, I think it should be declared in the configuration of each optimizer, a bit like weights decay is. The gradient clipper can be used standalone if someone wants to implement their own training loop, nobody is forcing gradient clipper to be used exclusively by optimizers, it's just a convenient configuration that can be turned on.

So I think we can keep the gradient clipper in the optimizer, it's convenient and doesn't really block any complex use case.